src/pal/pal_posix.h

```diff
@@ -264,5 +270,31 @@ namespace snmalloc
 
       OS::error("Out of memory");
     }
+
+    /**
+     * Source of Entropy
+     *
+     * This is a default that works on many POSIX platforms.
+     */
+    static uint64_t get_entropy64()
+    {
+      if constexpr (!pal_supports<Entropy, OS>)
+      {
+        // Derived Pal does not provide entropy.
+        return 0;
+      }
+      else if constexpr (OS::get_entropy64 != get_entropy64)
+      {
+        // Derived Pal has provided a custom definition.
+        return OS::get_entropy64();
+      }
+      else
+      {
+        uint64_t result;
+        if (getentropy(&result, sizeof(result)) != 0)
```

Comment:
genuine question, would arc4random_buf would work as fallback for unsupported systems since it s aiming userland allocation only anyway ?

Reply:
Anything is better than nothing.  I am not an expert on APIs for random, so I'll look at that one.

Reply:
My current plan was to use 
```
std::random_device
```
as the back-up.  Would that be okay on the platforms that aren't currently working.  So if I just disable the Pal feature flag like I have on Haiku for those platforms.

Reply:
fair enough. ;-)

Reply:
also was just thinking but that depends how far back you re willing to support older oses versions. getentropy was introduced at various timing, so depends if you stick to constexpr approach vs cmake+ifdef'ries

Reply:
I am not keen to go to far down the ifdef'ries, but will if strong demand.